simd/unit_tests/TestSIMD.cpp

```diff
@@ -379,6 +379,76 @@ inline void host_check_condition() {
   EXPECT_TRUE(all_of(a == decltype(a)(16)));
 }
 
+template <typename Abi, typename DataType>
+inline void host_check_where_expression_scatter_to() {
+  using simd_type  = Kokkos::Experimental::simd<DataType, Abi>;
+  using index_type = Kokkos::Experimental::simd<std::int32_t, Abi>;
+  using mask_type  = typename simd_type::mask_type;
+
+  std::size_t nlanes = simd_type::size();
+  mask_type mask(false);
+
+  for (std::size_t i = 0; i < nlanes; i += 2) {
+    mask[i] = true;
+  }
+
+  DataType simd_val = 40;
+  simd_type test_simd(simd_val);
+  DataType val   = 3;
+  DataType dst[] = {val, val, val, val, val, val, val, val};
+
+  index_type index(0);
```

Comment:
I think same index and value for all lane is not a good test.  I suspect the scale argument is wrong when you call the intrinsics and your test does not expose the bug(?)